api/envoy/api/v2/route/route_components.proto

```diff
@@ -1177,6 +1177,8 @@ message RedirectAction {
 
   oneof path_rewrite_specifier {
     // The path portion of the URL will be swapped with this value.
+    // Please note that query string in path_redirect will override the
+    // request's query string and will not be stripped.
```

Comment:
Can you add some examples here (make sure to add the examples to v3 also)? I've been looking at this for a few minutes and I'm a little confused as to what the current behavior is and what is changing.

Reply:
Currently the request's query string will be stripped regardless of the value of `strip_query`. 

With this patch, envoy would only strip the request's query string if `path_redirect` doesn't contain query string and `strip_query` is set to true. 
If `path_redirect` contains query string, users will be redirected to the specified path with query string no matter what the value of `strip_query` is.

Reply:
Thanks can you add some actual examples of real paths and what happens to them?